test/replication.test.js

```diff
@@ -1804,3 +1804,105 @@ describe('Replication / Change APIs', function() {
+      'SourceModelWithTenant-' + tid,
+      { id: { id: true, type: String, defaultFn: 'guid' } },
+      { trackChanges: true, tenantProperty: 'tenantId' });
+
+    SourceModelWithTenant.attachTo(dataSource);
+
+    TargetModelWithTenant = this.TargetModelWithTenant = PersistedModel.extend(
+      'TargetModelWithTenant-' + tid,
+      { id: { id: true, type: String, defaultFn: 'guid' } },
+      { trackChanges: true, tenantProperty: 'tenantId' });
+
+    var TargetWithTenantChange = TargetModelWithTenant.Change;
+    TargetWithTenantChange.Checkpoint = loopback.Checkpoint.extend('TargetWithTenantCheckpoint');
+    TargetWithTenantChange.Checkpoint.attachTo(dataSource);
+
+    TargetModelWithTenant.attachTo(dataSource);
+
+    var originalFilterFactory = SourceModelWithTenant.createChangeFilter;
+    SourceModelWithTenant.createChangeFilter = function(since, modelFilter) {
+      var tenantProperty = TargetModelWithTenant.settings.tenantProperty;
+
+      var filter = originalFilterFactory.call(this, since, modelFilter);
+
+      if (modelFilter && modelFilter.where && modelFilter.where[tenantProperty]) {
+        filter.where.tenant = modelFilter.where[tenantProperty];
+      }
+      return filter;
+    };
+
+    SourceModelWithTenant.prototype.fillChangeTenantValue = function(change, cb) {
+      var model = TargetModelWithTenant;
+      var tenant = null;
+      var tenantProperty;
+      var inst = this;
+
+      if (model && model.settings && model.settings.tenantProperty) {
+        tenantProperty = model.settings.tenantProperty;
+      }
+
+      if (tenantProperty && inst && inst[tenantProperty]) {
+        tenant = inst[tenantProperty];
+      }
+
+      change.tenant = tenant;
+
+      cb(null);
+    };
+
+    test.startingCheckpoint = -1;
+  });
+
+  describe('Model.changes(since, filter, callback) - tenant variant', function() {
+    var tenant = '123';
+
+    it('Get changes since the given checkpoint for the given tenant', function(done) {
+      var test = this;
+      this.SourceModelWithTenant.create([{name: 'foo', tenantId: tenant}, {name: 'foo', tenantId: '456'}], function(err) {
+        if (err) return done(err);
+
+        setTimeout(function() {
+          test.SourceModelWithTenant.changes(test.startingCheckpoint, {where: {tenantId: tenant}}, function(err, changes) {
+            assert.equal(changes.length, 1);
+            assert.equal(changes[0].tenant, tenant);
+
+            done();
+          });
+        }, 1);
+      });
+    });
+
+    it('excludes changes from older checkpoints', function(done) {
+      var FUTURE_CHECKPOINT = 999;
+
+      SourceModelWithTenant.create({ name: 'foo', tenantId: tenant }, function(err) {
+        if (err) return done(err);
+        SourceModelWithTenant.changes(FUTURE_CHECKPOINT, {where: {tenantId: tenant}}, function(err, changes) {
+          if (err) return done(err);
+
+          expect(changes).to.be.empty; //jshint ignore:line
```

Comment:
```js
expect(changes).to.be.empty();
```

Just make sure `expect` is defined as `var expect = require('./helpers/expect')` at the top of this file.